src/PowerShellEditorServices/Services/Symbols/Visitors/RegionVisitor.cs

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Management.Automation.Language;
+using Microsoft.PowerShell.EditorServices.Services.TextDocument;
+
+namespace Microsoft.PowerShell.EditorServices.Services.Symbols
+{
+    internal static class RegionVisitor
+    {
+        internal static void GetRegionsInDocument(ScriptFile file, Func<SymbolReference, AstVisitAction> action)
+        {
+            Stack<Token> tokenCommentRegionStack = new();
+            Token[] tokens = file.ScriptTokens;
+
+            for (int i = 0; i < tokens.Length; i++)
```

Comment:
Heh, I hate seeing this over `foreach` but I see `IsBlockComment` takes the index for some reason...

Reply:
It uses index to verify the previous token was a new line.